events/formatter.js

```diff
@@ -0,0 +1,52 @@
+const ics = require('ics');
+
+// TODO: migrate to a class if we're handling all those cases
+function toICS(result) {
+  let events = [];
+  if (result.results) {
+    events = result.results;
+  } else {
+    events = [result];
+  }
+  const icsEvents = events.map(e => ({
+    title: e.name,
+    description: e.description,
+    uid: e.id,
+    lastModified: e.updatedAt,
+    organizer: { name: 'CoderDojo', email: 'info@coderdojo.com' },
+    start: [
+      e.startTime.getUTCFullYear(),
+      e.startTime.getUTCMonth(),
+      e.startTime.getUTCDate(),
+      e.startTime.getUTCHours(),
+      e.startTime.getUTCMinutes(),
+      e.startTime.getUTCSeconds(),
+    ],
+    end: [
+      e.endTime.getUTCFullYear(),
+      e.endTime.getUTCMonth(),
+      e.endTime.getUTCDate(),
+      e.endTime.getUTCHours(),
+      e.endTime.getUTCMinutes(),
+      e.endTime.getUTCSeconds(),
+    ],
+    // TODO: Add applications if they are loaded
+    // TODO: Add status of the even depending on the applications
+    url: `https://zen.coderdojo.com/events/${e.id}`,
+  }));
+  return ics.createEvents(icsEvents).value;
+}
+module.exports.format = function format(req, res) {
+  let response;
+  res.format({
+    'text/calendar': () => {
```

Comment:
Where is the dispatch taking place? Will a single GET to the dojo url with an ".ics" at the end suffice or do I need to send headers?
I would prefer the .ics ending.

Reply:
Another PR I'll push soon. The events-service is back-office only, the accessible url (from cp-zen-platform) won't require the header but will indeed end in `.ics` as an optional formatting (which means the results from the json and the ics will be consistent without code duplication)